### DISCUSSION TITLE
Ollama embedding error context length exceeded for jina-embeddings-v2-zh

### DISCUSSION THREAD

**Discussion by chenyuz3** [logancyang/obsidian-copilot#1164] [Q&A]
I noticed that for no matter what models used for vualt QA embedding, the default context length/chunk size is 2048 tokens, which may lead to a reduction of retrieval performance as any note longer than that will be simply abandoned. Many embedding models have larger context window by default so may be we can add an option in settings for that?
![image](https://github.com/user-attachments/assets/9b7d0a88-a7b3-442d-8747-8905a3a41425)

**Comment by logancyang** [comment]
@chenyuz3 Ollama has model-specific settings which you can set using `/set parameter ...`. Embedding context length is model-specific, so it shouldn't be set on the client side imo.

**Comment by chenyuz3** [comment]
Thx @logancyang , I have created a new model based on the original model with a context length larger, however I can still see some errors stating the context window is still not enough for some notes, I wonder how do copilot coupe with such situation? Does it simply cut off the note or something else?
![image](https://github.com/user-attachments/assets/f06626fc-c7f5-42da-92c5-6785623bb6ad)

![image](https://github.com/user-attachments/assets/f7236aaf-ce05-4512-af6a-0088db4cdf70)


![image](https://github.com/user-attachments/assets/c31363a8-0b9a-4b7c-926f-b66f72331f65)
![image](https://github.com/user-attachments/assets/5c688474-5a2e-4c8d-bb3e-f59a947a6997)

**Comment by logancyang** [comment]
@chenyuz3 copilot does nothing when it exceeds the context length, the error you see should fail the chunk entirely so it won't appear in the index.

It is up to the backend to decide whether to truncate. In the past Ollama truncated the excessive length silently without error, now it seems it no longer does that, which is better.

What embedding model are you using and how long exactly is your context length? Copilot's chunk is 4000 chars, i.e. roughly 1000 tokens, should be short enough for most embedding models.

**Comment by chenyuz3** [comment]
> @chenyuz3 copilot does nothing when it exceeds the context length, the error you see should fail the chunk entirely so it won't appear in the index.
> 
> It is up to the backend to decide whether to truncate. In the past Ollama truncated the excessive length silently without error, now it seems it no longer does that, which is better.
> 
> What embedding model are you using and how long exactly is your context length? Copilot's chunk is 4000 chars, i.e. roughly 1000 tokens, should be short enough for most embedding models.

@logancyang I use jina-embeddings-v2-zh as the embedding model with 8192 context length. I used to use bge-m3 models which also had a 8192 context length though at that time there were no embedding error found.

**Comment by logancyang** [comment]
@chenyuz3 I see. Could be language related since it's Chinese? I will test with `jina-embeddings-v2-zh` and see what's up.

**Comment by chenyuz3** [comment]
> @chenyuz3 I see. Could be language related since it's Chinese? I will test with `jina-embeddings-v2-zh` and see what's up.

@logancyang The bge-m3 model I used to use is also multilingual (including Chinese). I should say there were one or two errors 
(code 400) when using it, though the copilot plugin did not say it was a context length issue.
